packages/components/button/src/button.vue

```diff
@@ -6,18 +6,26 @@
     :class="[
       `puik-button--${variant}`,
       `puik-button--${size}`,
-      { 'puik-button--disabled': disabled },
+      { 'puik-button--disabled': disabled || loading },
       { 'puik-button--fluid': fluid },
     ]"
-    :disabled="disabled"
+    :disabled="disabled || loading"
     @click="setSelected"
   >
     <puik-icon
-      v-if="leftIcon"
+      v-if="leftIcon && !loading"
       :icon="leftIcon"
       :font-size="size !== 'sm' ? '1.25rem' : '1rem'"
       class="puik-button__left-icon"
     />
+
+    <puik-spinner-loader
+      v-if="loading"
+      :size="size"
+      color="reverse"
+      class="puik-button__loading"
+    />
+
     <slot></slot>
     <puik-icon
```

Comment:
So the right icon stays if the loading is enabled ?

Reply:
Maybe I don't have this info

Reply:
OK ... Also the contrast with different variant doesn't seems enough, the loader is invisible on variant `text`